bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/GarbageCollectorThread.java

```diff
@@ -306,7 +312,11 @@ public void safeRun() {
         boolean suspendMajor = suspendMajorCompaction.get();
         boolean suspendMinor = suspendMinorCompaction.get();
 
-        runWithFlags(force, suspendMajor, suspendMinor);
+        try {
+            runWithFlags(force, suspendMajor, suspendMinor);
+        } catch (Exception e) {
+            LOG.warn("Failed to complete gc {}", e.getMessage(), e);
```

Comment:
if we are catching "Exception" any time you call runWithFlags....
why don't catch the error inside runWithFlags and leave the previous signature ?

catch (Exception) is a code smell, do we really need it ?